ably-ios/ARTAuth.m

```diff
@@ -271,39 +316,48 @@ + (NSString *)random {
     NSUInteger r2 = arc4random_uniform(100000000);
     return [NSString stringWithFormat:@"%08lu%08lu", (long)r1, (long)r2];
 }
-/* TODO
+
 + (ARTSignedTokenRequestCb)defaultSignedTokenRequestCallback:(ARTAuthOptions *)authOptions rest:(ARTRest *)rest {
-    NSString *keyId = authOptions.keyId;
-    NSString *keyValue = authOptions.keyValue;
-    BOOL queryTime = authOptions.queryTime;
+    
+    //TODO should be block i think.
+     NSString *keyId = authOptions.keyId;
+     NSString *keyValue = authOptions.keyValue;
+     BOOL queryTime = authOptions.queryTime;
+    
+    NSString * theClientId = authOptions.clientId;
 
     __weak ARTRest *weakRest = rest;
 
     NSAssert(keyId && keyValue, @"keyId and keyValue must be set when using the default token auth");
 
-    return ^(ARTAuthTokenParams *params, void(^cb)(NSString *)) {
-        if (params.id && ![params.id isEqualToString:keyId]) {
-            NSLog(@"Params id is not equal to authOptions id");
+    return ^id<ARTCancellable>(ARTAuthTokenParams *params, void(^cb)(ARTAuthTokenParams *)) {
+        
+        
+    
+        if (params.keyName && ![params.keyName isEqualToString:keyId]) {
+            [ARTLog error:[NSString stringWithFormat:@"ARTAuth params keyname %@ is not equal to authOptions id %@", params.keyName, keyId]];
             cb(nil);
-            return;
+            return nil;
         }
-
+//TODO RM reqObj I think.
```

Comment:
Again, not too comfortable with comments that say "... I think". Can we figure out the right solution and add a test to ensure it is right?